### PR TITLE
Don't require clippy/miri for beta

### LINKED ISSUE #50557: Clippy and Miri should not block beta releases
I think this is the fourth beta I'm having to re-figure-out how all this works and figure out how to ignore their failures on beta. @kennytm can we just exclude these tools from toolstate until this is figured out?

### PR DESCRIPTION
**Pull request by oli-obk**
r? @kennytm 

cc @alexcrichton 

I'm trying this out locally atm to see if it works as I think it should. Not sure how to test it for real except wait for the next beta.

fixes #50557

### CODE REVIEW

File: src/ci/docker/x86_64-gnu-tools/checktools.sh [hunk 1]
```diff
@@ -57,17 +57,35 @@ verify_status() {
     fi
 }
 
+check_dispatch() {
+    if [ "$1" = submodule_changed ]; then
+        # ignore $2 (branch id)
+        verify_status $3 $4
+    else
+        if [ "$2" = beta ]; then
```
Comment (kennytm) [inline src/ci/docker/x86_64-gnu-tools/checktools.sh]: This could be `elif`

File: src/ci/docker/x86_64-gnu-tools/checktools.sh [hunk 2]
```diff
@@ -57,17 +57,35 @@ verify_status() {
...
+            if grep -vq '"'"$1"'":"(test|build)-fail"' "$TOOLSTATE_FILE"; then
```
Comment (kennytm) [inline src/ci/docker/x86_64-gnu-tools/checktools.sh]: 1. You should just use `-q` here. The `-v` flag means inverted search 
2. `grep` by default uses BRE which means you need to escape the brackets and vertical bars:

    ```sh
    if grep -q '"'"$1"'":"\(test\|build\)-fail"' "$TOOLSTATE_FILE"; then
    ```

    or use `-E` to use ERE.

File: src/ci/docker/x86_64-gnu-tools/checktools.sh (hunk 1 above)
Comment (alexcrichton) [reply src/ci/docker/x86_64-gnu-tools/checktools.sh]: I think "beta" also here may want to be "beta_required"?

**Comment by alexcrichton** [comment]
Awesome, thanks @oli-obk!

**Comment by oli-obk** [comment]
Locally I'm getting

```
Verifying status of book...
Verifying status of nomicon...
Verifying status of reference...
Verifying status of rust-by-example...
Verifying status of rls...
Verifying status of rustfmt...
Verifying status of clippy-driver...
Verifying status of miri...
Requiring test passing for book...
Requiring test passing for nomicon...
Requiring test passing for reference...
Requiring test passing for rust-by-example...
Requiring test passing for rls...
Requiring test passing for rustfmt...
```

if not on the nightly channel.

File: src/ci/docker/x86_64-gnu-tools/checktools.sh (hunk 1 above)
Comment (oli-obk) [reply src/ci/docker/x86_64-gnu-tools/checktools.sh]: so close, I tried `elsif`

File: src/ci/docker/x86_64-gnu-tools/checktools.sh (hunk 2 above)
Comment (oli-obk) [reply src/ci/docker/x86_64-gnu-tools/checktools.sh]: done

**Comment by kennytm** [comment]
@bors r+

Thanks!

**Comment by kennytm** [comment]
🤔 bors is not seeing this PR. Closing and reopening.

File: src/ci/docker/x86_64-gnu-tools/checktools.sh (hunk 1 above)
Comment (oli-obk) [reply src/ci/docker/x86_64-gnu-tools/checktools.sh]: the "beta" here is the second arg to all function calls in status_check. It denotes the branch, not the kind of check. I should probably add more comments

**Pull request merged**